jwst/ami/utils.py

```diff
@@ -285,3 +288,99 @@ def findmax(mag, vals, mid=1.0):
     maxx, maxy, fitc = quadratic(p, fitr)
 
     return maxx, maxy
+
+
+def pix_median_fill_value(input_array, input_dq_array, bsize, xc, yc):
+    """
+    Short Summary
+    -------------
+    For the pixel specified by (xc, yc), calculate the median value of the
+    good values within the box of size bsize neighboring pixels. If any of
+    the box is outside the data, 0 will be returned.
+
+    Parameters
+    ----------
+    input_array: ndarray
+        2D input array to filter
+    input_dq_array: ndarray
+        2D input data quality array
+    bsize: scalar
+        square box size of the data to extract
+    xc: scalar
+        x position of the data extraction
+    yc: scalar
+        y position of the data extraction
+
+    Returns
+    -------
+    median_value: float
+        median value of good values within box of neighboring pixels
+
+    """
+    # set the half box size
+    hbox = int(bsize/2)
+
+    # Extract the region of interest for the data
+    try:
+        data_array = input_array[xc - hbox:xc + hbox, yc - hbox: yc + hbox]
+        dq_array = input_dq_array[xc - hbox:xc + hbox, yc - hbox: yc + hbox]
+    except IndexError:
+        # If the box is outside the data return 0
+        log.warning('Box for median filter is outside the data.')
+        return 0.
+
+    filtered_array = data_array[dq_array != dqflags.pixel['DO_NOT_USE']]
```

Comment:
Hmmm, not exactly sure how equality comparisons turn out when we're interested in the settings of individual bits, but I'm worried that simply using `!=` will only exclude pixels that have a DQ value exactly equal to `DO_NOT_USE`. But in most cases there's also another informational DQ value set along with `DO_NOT_USE`, in order to indicate why that pixel is bad. So will this statement correctly exclude pixels that have multiple flags set along with `DO_NOT_USE`?  Might need to use one of the bitwise comparators in some way.

Reply:
Perhaps something like:
```
goodpixels = np.where(np.bitwise_and(dq, dqflags.pixel['DO_NOT_USE']) == 0)
```

Reply:
The bitwise check does it, irrespective of other DQ bits.